pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -165,17 +166,30 @@ func (c *BootstrapTeardownController) canRemoveEtcdBootstrap() (bool, bool, erro
 		return false, hasBootstrap, nil
 	}
 
-	if len(members) < 4 {
+	isUnsupportedUnsafeEtcd, err := c.isUnsupportedUnsafeEtcd()
+	if err != nil {
+		return false, hasBootstrap, err
+	}
+
+	switch {
+	case !isUnsupportedUnsafeEtcd && len(members) < 4:
```

Comment:
comment the cases or you'll forget what you're trying to do

Reply:
added the comments